tests/components/sonos/test_init.py

```diff
@@ -89,14 +89,21 @@ async def test_async_poll_manual_hosts_warnings(
     ), patch.object(
         hass, "async_add_executor_job", new=AsyncMock()
     ) as mock_async_add_executor_job:
```

Comment:
I think that `get_sync_attributes` should be moved to `helpers.py`
Then you can patch that method instead of `async_add_executor_job` which is too broad IMO:
```suggestion
    ), patch(
        "homeassistant.components.sonos.get_sync_attributes",
        side_effect=[
            OSError(),
            OSError(),
            [],
            [],
            OSError(),
        ],
    ):
```

Reply:
On the other hand, it prevents from thread context switching 15-times for that single test - which is just testing logger calls 🤷

Reply:
Readability I think is more important than performance here in the tests.

Reply:
Oh, I guess I can just patch `soco.visible_zones`. That way we wouldn't have to patch any integration code, but only lib code.  
I have been told not to patch integrations code in tests previously from core members.

Reply:
> Oh, I guess I can just patch `soco.visible_zones`. That way we wouldn't have to patch any integration code, but only lib code. I have been told not to patch integrations code in tests previously from core members.

That would be best indeed but the other properties also seem to make calls. I think patching `get_sync_attributes` isn't perfect but it's better than patching `async_add_executor_job`